src/platform/ESP32/bluedroid/BLEManagerImpl.cpp

```diff
@@ -768,6 +810,7 @@ void BLEManagerImpl::HandleGATTControlEvent(esp_gatts_cb_event_t event, esp_gatt
             mFlags.Set(Flags::kAppRegistered);
             controlOpComplete = true;
         }
+        esp_ble_gap_config_local_privacy(true);
```

Comment:
Previously the spec mentioned to change the address between every commissioning attempt. But, with the latest changes, it says that the random device address shall change atleast on every boot.

This is done in register event and will follow the new/latest requirement. Although, in NimBLE for now, we change the address between every commissioning attempts. I will change that in a subsequent PR